pkg/controller/operconfig/migration.go

```diff
@@ -97,3 +132,388 @@ func convertEgressFirewallToEgressNetworkPolicy(ctx context.Context, client cnoc
+
+		if err := deleteCloudPrivateIpConfigs(ctx, client, egressIps); err != nil {
+			return err
+		}
+
+		eipNamespace, found, err := unstructured.NestedString(eip.Object, "spec", "namespaceSelector", "matchLabels", "kubernetes.io/metadata.name")
+		if !found || err != nil {
+			return fmt.Errorf("kubernetes.io/metadata.name not found in egressIP object %s, probable underlying error: %v", eip.GetName(), err)
+		}
+
+		if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+			nns, err := client.Default().Dynamic().Resource(gvrNetnamespace).Get(ctx, eipNamespace, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+
+			nns.Object["egressIPs"] = egressIps
+			_, err = client.Default().Dynamic().Resource(gvrNetnamespace).Update(ctx, nns, metav1.UpdateOptions{})
+			return err
+		}); err != nil {
+			return err
+		}
+	}
+
+	// 4. iterate through nodes and generate hostsubnets for any that are egress-assignable
+	for _, node := range nodeList.Items {
+		if node.Labels == nil {
+			continue
+		}
+		if _, ok := node.Labels[egressAssignable]; ok {
+			// generate egressCIDRs field
+			// if "egressCIDRAnnotationName" node annotation exists, automatic sdn config was used and we shall reuse old config
+			// else, manual sdn config was used and we shall use node's subnet CIDR (cannot restore egressIPs)
+			var egressCIDRsArr []string
+			if _, ok := node.Annotations[egressCIDRAnnotationName]; ok {
+				ovnMigrationNodeAnnotation := OVNMigrationNodeAnnotation{}
+				if err := json.Unmarshal([]byte(node.Annotations[egressCIDRAnnotationName]), &ovnMigrationNodeAnnotation); err != nil {
+					return err
+				}
+				egressCIDRsArr = ovnMigrationNodeAnnotation.EgressCIDRs
+			} else {
+				egressIpConfig := node.Annotations[egressIPNodeConfig]
+				egressIpConfigArr := make([]NodeEgressIpConfig, 0)
+				if err := json.Unmarshal([]byte(egressIpConfig), &egressIpConfigArr); err != nil {
+					return err
+				}
+				nodeSubnet := egressIpConfigArr[0].IfAddr["ipv4"]
+				egressCIDRsArr = []string{nodeSubnet}
+			}
+
+			if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+				hsn, err := client.Default().Dynamic().Resource(gvrHostSubnet).Get(ctx, node.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+
+				hsn.Object["egressCIDRs"] = ConvertToUnstructuredInterface(egressCIDRsArr)
+				_, err = client.Default().Dynamic().Resource(gvrHostSubnet).Update(ctx, hsn, metav1.UpdateOptions{})
+				return err
+			}); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil // success
+}
+
+func sdnEgressIpResourcesReady(ctx context.Context, client cnoclient.Client) (bool, error) {
+	// get all nodes
+	// get all hostsubnets
+	// iterate over all node names and return false if any of them don't have an associated hostsubnet
+	// here!!
+	nodeList, err := client.Default().Kubernetes().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
```

Comment:
From Dan Winship:
"(FWIW defaultPageSize is 500 and we don't even support clusters with that many nodes so this will basically always be equivalent to doing a non-paged List.)"

Reply:
mkay, I was thinking on a 50 default page size for some reason

Reply:
I think I'll leave this as is for now just to keep it consistent between my different PRs...I'm definitely open to changing this later if the need arises; it's not a difficult change to make.